### DISCUSSION TITLE
Reading query params from the URL

### DISCUSSION THREAD

**Discussion by alpkaanaksu** [rubenarslan/formr.org#558] [Q&A] [answered]
Hi Ruben,
what is the best way to access the query params from the URL?

e.g. `https://formr.instance.com/run?message=hey`

How can we access the character value `"hey"`? How can we save this information permanently? (so that we can write something like `params$message` and get `"hey"`)

**Comment by alpkaanaksu** [comment]
I think I found it here: https://github.com/rubenarslan/formr.org/wiki/How-to-collect-an-ID-(Mturk-Prolific-etc.)-from-the-URL

**Reply by alpkaanaksu** [reply]
@rubenarslan Ok, it kind of works with the `get param` type, but as soon as we click on a submit button, or go to next page, the page reloads and we are in a new session, the URL is the run base URL without the query params.

**Reply by rubenarslan** [reply]
The value of the get parameter should be stored within the survey table at that point. Not so?

**Reply by alpkaanaksu** [reply]
<img width="1014" alt="Screenshot 2024-06-20 at 14 57 23" src="https://github.com/rubenarslan/formr.org/assets/68744864/0bfe895a-8992-4c7c-bb28-b3b0f44e1552">

They are saved in one of the sessions, but then we are in a new session somehow

**Reply by alpkaanaksu** [reply]
It works at first, but as soon as I click on submit the page reloads instead of going to the next page, and I get an error:

![IMG_169CDF18000A-1](https://github.com/rubenarslan/formr.org/assets/68744864/50a1b832-3cee-448f-80b5-c023743202cf)

**Comment by rubenarslan** [comment]
Please make a reproducible example, ie a minimal run with a survey showing
the problem.

**Reply by alpkaanaksu** [reply]
We couldn't create a minimal run because it works when we create a new run with a shortened version of the survey. I tried duplicating the run (by exporting and reimporting) and now it works in the new run. Very strange behaviour... But it works now (but only in the new run) 🤔

**Reply by rubenarslan** [reply]
That's odd. Try clearing cookies for the formr subdomains on your instance or opening the run in an incognito tab. That should not be necessary, but if this helps, I'll have an idea about how to keep this from recurring. The problem wouldn't affect your participants in that case though.

**Reply by alpkaanaksu** [reply, marked as answer]
I was doing all of this in incognito mode, so it was not a problem with the browser cache.

We were trying to access the run on `formr.osc.lmu.de` instead of on `surveys.osc.lmu.de`, so after opening the page, we were directed to `surveys.osc.lmu.de`. This is probably a thing with the configuration of our instance. Now it is working

**Reply by rubenarslan** [reply]
If you were doing _all_ of it an incognito tab (including editing the run), there could be a complication. Yes, studies are logically separated from the admin area on your instance. The link to the run displayed is with the surveys subdomain. Access to the run from the formr subdomain should be blocked, but I see that it isn't. I'll check it out.
